scilpy/segment/recobundlesx.py

```diff
@@ -135,66 +144,55 @@ def _cluster_model_bundle(self, model, model_clust_thr, identifier=None):
     def _reduce_search_space(self, neighbors_reduction_thr=18):
         """
         Wrapper function to discard clusters from the tractogram too far from
-        the model and logging informations
+        the model and logging informations.
         :param neighbors_reduction_thr, float, distance in mm for thresholding
-            to discard distant streamlines
+            to discard distant streamlines.
         """
         centroid_matrix = bundles_distances_mdf(self.model_centroids,
                                                 self.centroids)
         centroid_matrix[centroid_matrix >
                         neighbors_reduction_thr] = np.inf
 
         mins = np.min(centroid_matrix, axis=0)
-        close_clusters_indices = list(np.where(mins != np.inf)[0])
-        if len(close_clusters_indices) < 1:
+        close_clusters_indices = np.array(np.where(mins != np.inf)[0],
+                                          dtype=np.int32)
+
+        if not len(close_clusters_indices):
```

Comment:
You could remove lines 160, 161 and then at line 173
```
return len(close_clusters_indices) > 0
```
?

Would be a more meaningful return and I think the rest of the code would handle the situation well, loops just wouldn't be branched into